src/System.Net.Http.Json/src/System/Net/Http/Json/HttpContentJsonExtensions.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.IO;
+using System.Net.Mime;
+using System.Text;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Net.Http.Json
+{
+    public static class HttpContentJsonExtensions
+    {
+        public static Task<object?> ReadFromJsonAsync(this HttpContent content, Type type, JsonSerializerOptions? options = null, CancellationToken cancellationToken = default)
+        {
+            return ReadFromJsonAsyncCore(content, type, options, cancellationToken);
+        }
+
+        public static Task<T> ReadFromJsonAsync<T>(this HttpContent content, JsonSerializerOptions? options = null, CancellationToken cancellationToken = default)
+        {
+            return ReadFromJsonAsyncCore<T>(content, options, cancellationToken);
+        }
+
+        private static async Task<object?> ReadFromJsonAsyncCore(HttpContent content, Type type, JsonSerializerOptions? options, CancellationToken cancellationToken)
+        {
+            Stream contentStream = await GetJsonStreamFromContentAsync(content).ConfigureAwait(false);
+            return await JsonSerializer.DeserializeAsync(contentStream, type, options, cancellationToken);
+        }
+
+        private static async Task<T> ReadFromJsonAsyncCore<T>(HttpContent content, JsonSerializerOptions? options, CancellationToken cancellationToken)
+        {
+            Stream contentStream = await GetJsonStreamFromContentAsync(content).ConfigureAwait(false);
+            return await JsonSerializer.DeserializeAsync<T>(contentStream, options, cancellationToken).ConfigureAwait(false);
+        }
+
+        private static async Task<Stream> GetJsonStreamFromContentAsync(HttpContent content)
+        {
+            ValidateMediaType(content.Headers.ContentType?.MediaType);
+            Debug.Assert(content.Headers.ContentType != null);
```

Comment:
If we're able to assert that ContentType != null, why do we use the null-coalescing operator on the previous line?  And is this assert actually valid... what guarantees ContentType is non-null?

Reply:
`ValidateMediaType` throws if the mediaType argument is null, it will be null if ContentType is null, it will be "application/json" of "text/plain" if is not null.

Reply:
Ok